python-backend/app/api/mines/compliance/resources/compliance.py

```diff
@@ -33,33 +33,59 @@ def get(self, mine_no=None):
 
             advisories = 0
             warnings = 0
-            open_orders = 0
-            overdue_orders = 0
+            num_open_orders = 0
+            num_overdue_orders = 0
             section_35_orders = 0
+            open_orders_list = []
 
             for report in data:
-                report_date = datetime.strptime(report.get('assessmentDate'), '%Y-%m-%d %H:%M')
+                report_date = self.get_datetime_from_NRIS_data(report.get('assessmentDate'))
                 one_year_ago = datetime.now() - relativedelta(years=1)
 
                 inspection = report.get('inspection')
                 stops = inspection.get('stops')
+                order_count = 1
 
-                if stops:
-                    stops_dict = stops[0]
+                for stop in stops:
 
-                    stop_orders = stops_dict.get('stopOrders')
-                    stop_advisories = stops_dict.get('stopAdvisories')
-                    stop_warnings = stops_dict.get('stopWarnings')
+                    stop_orders = stop.get('stopOrders')
+                    stop_advisories = stop.get('stopAdvisories')
+                    stop_warnings = stop.get('stopWarnings')
 
-                    if stop_orders:
-                        open_orders += sum(k.get('orderStatus') == 'Open' for k in stop_orders)
-                        overdue_orders += sum(
-                            k.get('orderCompletionDate') is not None
-                            and k.get('orderStatus') == 'Open' and datetime.strptime(
-                                k.get('orderCompletionDate'), '%Y-%m-%d %H:%M') < datetime.now()
-                            for k in stop_orders)
-                        section_35_orders += sum(
-                            k.get('orderAuthoritySection') == 'Section 35' for k in stop_orders)
+                    for order in stop_orders:
+                        if order.get('orderStatus') == 'Open':
+
+                            legislation = order.get('orderLegislations')
+                            permit = order.get('orderPermits')
+                            section = None
+
+                            if legislation:
+                                section = legislation[0].get('section')
+                            elif permit:
+                                section = permit[0].get('permitSectionNumber')
+
+                            order_to_add = {
+                                'order_no': f'{report.get("assessmentId")}-{order_count}',
+                                'code_viloation': section,
+                                'report_no': report.get('assessmentId'),
+                                'inspector': report.get('assessor'),
+                                'due_date': order.get('orderCompletionDate'),
+                                'overdue': False,
+                            }
+
+                            num_open_orders += 1
+
+                            if order.get('orderCompletionDate'
```

Comment:
This is the sort of line break that the formatter is adding that I don't think is appropriate. It's so difficult to read, and the break seems really arbitrary

Earlier in this file we have:
```python
return self.raise_error(errhttp.response.status_code,
                                     'NRIS error occurred.' + str(errhttp))
```

Which has the closing `)` on the same line. This break has the closing `)` on a new line. Then the check after the `and` has the opening `(` on its own line.

Reply:
I manually tried to break this in a more readable better way. The formatter doesn't always make the most readable choices.